Compiler warning about null-pointer exception detection on un-mapped flash areas when `CONFIG_NULL_POINTER_EXCEPTION_DETECTION_MPU=y` on `mimxrt685_evk/mimxrt685s/cm33`

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**
<!--

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - ...
-->
Following compiler warning occurs when I3C device with 9 byte of reg address to dts, and DCONFIG_NULL_POINTER_EXCEPTION_DETECTION_MPU=y.
_C:/repos/ADI-zephyr/zephyr/arch/arm/core/mpu/arm_mpu.c: In function 'z_arm_mpu_init':_
_C:/repos/ADI-zephyr/zephyr/arch/arm/core/mpu/arm_mpu.c:481:8: warning: integer constant is too large for its type_
  _481 |         (!DT_NULL_PAGE_DETECT_NODE_EXIST)_


- I3C device requires 9 byte of address according to i3c-device.yaml 
https://git.byronlathi.com/bslathi19/zephyr/-/blob/main/dts/bindings/i3c/i3c-device.yaml?ref_type=heads

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->
1. Build a board with M33 with DCONFIG_NULL_POINTER_EXCEPTION_DETECTION_MPU=y with a tests/*:
example:
west build -b mimxrt685_evk/mimxrt685s/cm33 .\tests\drivers\uart\uart_basic_api\ -p always -DCONFIG_NULL_POINTER_EXCEPTION_DETECTION_MPU=y
2. See the compiler warning:
_warning: integer constant is too large for its type
  481 |         (!DT_NULL_PAGE_DETECT_NODE_EXIST)_
  
**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->
Compiler warning shall not occur.
**Impact**
<!--
What impact does this issue have on your progress (e.g., annoyance, showstopper)
-->
CI/CD requires -Werror option, thus this fails CI/CD procedure in out-of-tree zephyr pipeline.
**Logs and console output**
<!--
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)
-->

**Environment (please complete the following information):**
 - Zephyr 4.00+ (macro DT_NULL_PAGE_DETECT_NODE_EXIST added [here](https://github.com/zephyrproject-rtos/zephyr/commit/f2e115cca374516cb9e2a5845d1e3b77b53a18c9#diff-c7c1d65647254c45a9882753a577c36e14758fd0fd77f689504cea5cc03d4a65R44) )
 - OS: all
 - Toolchain Zephyr SDK
 - Commit SHA or Version used

There are 2 issues being discussed here:

----------------------------------------------------------------
#### 1st is the one reported by @mertekren which is this:


```
/zephyrproject/zephyr/arch/arm/core/mpu/arm_mpu.c:481:8: warning: integer constant is too large for its type
  481 |         (!DT_NULL_PAGE_DETECT_NODE_EXIST)
      | ^       ~~~~~~~~~~~~~~~~~~~~~
```

You see this warning because of the address of wm8904 in devicetree is too large:
```
	audio_codec: wm8904@1a0000000000000000 {
		compatible = "wolfson,wm8904";
		reg = <0x1a 0 0>;
                ...
	};

```
which results in this:
`#define DT_N_S_soc_S_peripheral_50000000_S_i3c_36000_S_wm8904_1a0000000000000000_REG_IDX_0_VAL_ADDRESS 479615345916448342016 /* 0x1a0000000000000000 */`

Since the `*nodeid_IDX_0_VAL_ADDRESS` has a large value, the `DT_FOREACH_STATUS_OKAY_NODE_VARGS()` used in `DT_NULL_PAGE_DETECT_NODE_EXIST` throws the compiler warning.

Initially, this didn't look like a DT macro issue to me 'coz `DT_REG_ADDR()` inside any `DT_FOREACH_*()` macro will throw the same compiler warning for this large value if used in any other code not just the one used in discussion. But then I found that similar the concern was raised, and the answer [here](https://github.com/zephyrproject-rtos/zephyr/pull/78445#issuecomment-2356320938) was that the large address is correct. So, I would like to kindly get feedback from @decsny and @rettichschnidi on how to address this. Is this an issue with the DT tooling?



---------------------------------------------------------------
##### The 2nd issue that @dcpleung mentioned is w.r.t. below:


```
/zephyrproject/zephyr/arch/arm/core/mpu/arm_mpu.c: In function 'z_arm_mpu_init':
/zephyrproject/zephyr/arch/arm/core/mpu/arm_mpu.c:483:9: note: '#pragma message: Null-Pointer exception detection cannot be configured on un-mapped flash areas'
  483 | #pragma message "Null-Pointer exception detection cannot be configured on un-mapped flash areas"
      |         ^~~~~~~

```

[This ](https://github.com/zephyrproject-rtos/zephyr/commit/f2e115cca374516cb9e2a5845d1e3b77b53a18c9) commit provided a solution to fix this. If the board wants to have mpu config for null point dereference then they can define a node as mentioned in the commit.
@mertekren, I would suggest creating another issue for this 2nd one and assigning it to the respective board owner.

--- a/arch/arm/core/mpu/arm_mpu.c
+++ b/arch/arm/core/mpu/arm_mpu.c
@@ -42,7 +42,8 @@ BUILD_ASSERT((DT_FOREACH_STATUS_OKAY_NODE_VARGS(
 	 (DT_REG_SIZE(node_id) >= CONFIG_CORTEX_M_NULL_POINTER_EXCEPTION_PAGE_SIZE)) ||
 
 #define DT_NULL_PAGE_DETECT_NODE_EXIST                                                             \
-	(DT_FOREACH_STATUS_OKAY_NODE_VARGS(NULL_PAGE_DETECT_NODE_FINDER, zephyr_memory_attr) false)
+	(DT_FOREACH_STATUS_OKAY_VARGS(zephyr_memory_region, NULL_PAGE_DETECT_NODE_FINDER,          \
+				      zephyr_memory_attr) false)
 
 /*
  * Global status variable holding the number of HW MPU region indices, which

